### PR TITLE
Adds migration guide from deprecated splash screen API

### PR DESCRIPTION
**Pull request by camsim99**
Addresses https://github.com/flutter/flutter/issues/105173.

Adds a migration guide from defining a custom splash screen to defining a launch screen that will persist until the Flutter UI is loaded, and updates docs referencing splash screens across the site.

Also links docs to splash screen sample app and addresses https://github.com/flutter/website/issues/7151.

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [ ] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Comment by khanhnwin** [comment]
PTAL @sfshaza2!

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
Hi Camille! Thanks for this PR!

However, there are a couple issues. This isn't where the migration guide goes. It's probably easier if we meet via GVC and I can explain. I promise to remember you.  ;)

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
some edits...

**Review by khanhnwin — APPROVED** [review]

**Comment by sfshaza2** [comment]
This is not ready to land. I met again with @camsim99 today to discuss more changes, namely:
* She's also going to update the original page to link to the migration page.
* A few more edits to the new page, once we talked through the impact of the but that causes crashing for apps created *before* Flutter 2.5, but trying to run them on Flutter 2.5 or later.
* She's also going to go through the splash screen issues on the repo and address/close those.
* Finally, she's going to update references in the add-to-app docs to say "launch screen" instead of "splash screen", as we no longer implement splash screens by default.

**Comment by khanhnwin** [comment]
@sfshaza2 Sounds good. This PR won't be landed until your changes are LGTMd.

File: src/development/ui/advanced/splash-screen.md [hunk 1]
```diff
@@ -39,11 +45,21 @@ part of the [Human Interface Guidelines][].
...
+
 In Android, there are two separate screens that you can control:
 a _launch screen_ shown while your Android app initializes,
 and a _splash screen_ that displays while the Flutter experience
 initializes.
 
+As of Flutter 2.5, Flutter has consolidated these two screens
+into one Android launch screen. Flutter automatically keeps this
+launch screen displayed until it draws the first frame.
+
```
Comment (camsim99) [inline src/development/ui/advanced/splash-screen.md:62]: @sfshaza2 Maybe this should be emphasized?

**Comment by camsim99** [comment]
@sfshaza2 I went through the list of tasks, so this is ready for review.

I would like to note, though, that splash screen is a synonym for launch screen, so I didn't replace all of the usages of splash screen with launch screen. This is primarily because splash screen is an Android term, so I figure people may still be searching for "splash" screen versus "launch" screen. Their equivalent meanings are noted in all of the relevant guides now, so we just need to make sure it is clear in the migration guide.

**Comment by sfshaza2** [comment]
Ping me when you want me to do another review.  :)

File: src/development/add-to-app/android/add-flutter-fragment.md [hunk 2]
```diff
@@ -326,9 +326,9 @@ The initial display of Flutter content requires some wait time,
 even if a pre-warmed `FlutterEngine` is used.
 To help improve the user experience around
 this brief waiting period, Flutter supports the
-display of a splash screen until Flutter renders
-its first frame. For instructions about how to show a splash screen,
-see the [splash screen guide][].
+display of a splash screen (also known as "launch screen") until Flutter
+renders its first frame. For instructions about how to show a lanch
```
Comment (sfshaza2) [inline src/development/add-to-app/android/add-flutter-fragment.md:330]:
```suggestion
renders its first frame. For instructions about how to show a launch
```

File: src/development/platform-integration/android/splash-screen-migration.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+   public SplashScreen provideSplashScreen() {
+       // ...
+       return new DrawableSplashScreen(
+           new SomeDrawable(
+               ContextCompat.getDrawable(this, R.some_splash_screen)));
+   }
+   ```
+
+   In the next step, you'll specify this `Drawable` as your application's
+   launch theme, so delete this implementation.
```
Comment (sfshaza2) [inline src/development/platform-integration/android/splash-screen-migration.md:38]: Since the previous step now says to delete it, can we delete this step?

**Review by sfshaza2 — APPROVED** [review]
This is becoming a bit hard to follow. Maybe we can land it and then I can see it with fresh eyes?

File: src/development/ui/advanced/splash-screen.md (hunk 1 above)
Comment (camsim99) [reply src/development/ui/advanced/splash-screen.md:68]: I deleted the last paragraph because that isn't really true. You have to just use the launch screen from here on out.

The second paragraph is mentioned in the warning before, did you want to keep both?

**Comment by khanhnwin** [comment]
Landing it as Shams said! Thanks @camsim99!
